BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -1,294 +1,298 @@
+                                unsigned int id,
+                                const std::function<void(EventLogItem)>& callback,
+                                const std::vector<XpathQuery>& filters) {
+        auto query = std::wstring(L"Event/System[EventID=") + std::to_wstring(id) + std::wstring(L"]");
+        for(auto param : filters)
+            query += L" and " + param.ToString();
+
+        subscriptions.emplace_back(EventSubscription{ callback });
+        auto& eventSub = subscriptions[subscriptions.size() - 1];
+
+        EventWrapper hSubscription = EvtSubscribe(NULL, NULL, pwsPath.c_str(), query.c_str(), NULL,
+                                                  &subscriptions[subscriptions.size() - 1], CallbackWrapper,
+                                                  EvtSubscribeToFutureEvents);
+        eventSub.setSubHandle(hSubscription);
+
+        if(!hSubscription) {
+            if(ERROR_EVT_CHANNEL_NOT_FOUND == GetLastError())
+                LOG_WARNING("EventLogs::QueryEvents: Unable to find channel " << pwsPath);
+            else if(ERROR_EVT_INVALID_QUERY == GetLastError())
+                LOG_ERROR(L"EventLogs::SubscribeToEvent: query " << query << L" is not valid.");
+            else
+                LOG_ERROR("EventLogs::SubscribeToEvent: EvtSubscribe failed with " << GetLastError());
+
+            return std::nullopt;
+        }
+
+        return eventSub;
+    }
+
+    bool IsChannelOpen(const std::wstring& channel) {
+        PEVT_VARIANT pProperty = NULL;
+        PEVT_VARIANT pTemp = NULL;
+        DWORD dwBufferSize = 0;
+        DWORD dwBufferUsed = 0;
+
+        // Open the channel config
+        EventWrapper hChannel{ EvtOpenChannelConfig(NULL, channel.c_str(), 0) };
+        if(NULL == hChannel) {
+            LOG_ERROR(L"EventLogs::IsChannelOpen: EvtOpenChannelConfig failed with " + std::to_wstring(GetLastError()) +
+                      L" for channel " + channel);
+            return false;
+        }
+
+        // Attempt to get the channel property
+        if(!EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed)) {
+            auto status{ GetLastError() };
+            if(ERROR_INSUFFICIENT_BUFFER == status) {
+                dwBufferSize = dwBufferUsed;
+                pTemp = (PEVT_VARIANT) realloc(pProperty, dwBufferSize);
+
+                if(pTemp) {
+                    pProperty = pTemp;
+                    pTemp = NULL;
+                    EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty,
+                                                &dwBufferUsed);
+                } else {
+                    if(pProperty)
+                        free(pProperty);
+
+                    LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+                    return false;
+                }
+            }
+
```

Comment:
Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)